src/trace/context/extractors/http.spec.ts

```diff
@@ -0,0 +1,294 @@
+import { TracerWrapper } from "../../tracer-wrapper";
+import { HTTPEventSubType, HTTPEventTraceExtractor } from "./http";
+
+let mockSpanContext: any = null;
+
+// Mocking extract is needed, due to dd-trace being a No-op
+// if the detected environment is testing. This is expected, since
+// we don"t want to test dd-trace extraction, but our components.
+const ddTrace = require("dd-trace");
+jest.mock("dd-trace", () => {
+  return {
+    ...ddTrace,
+    _tracer: { _service: {} },
+    extract: (_carrier: any, _headers: any) => mockSpanContext,
+  };
+});
+const spyTracerWrapper = jest.spyOn(TracerWrapper.prototype, "extract");
+
+describe("HTTPEventTraceExtractor", () => {
+  describe("extract", () => {
+    beforeEach(() => {
+      mockSpanContext = null;
+    });
+
+    afterEach(() => {
+      jest.resetModules();
+    });
+
+    it("extracts trace context from payload with headers", () => {
+      mockSpanContext = {
+        toTraceId: () => "797643193680388254",
+        toSpanId: () => "4726693487091824375",
+        _sampling: {
+          priority: "2",
+        },
+      };
+      const tracerWrapper = new TracerWrapper();
+
+      const payload = {
+        headers: {
+          "x-datadog-parent-id": "4726693487091824375",
+          "x-datadog-sampling-priority": "2",
+          "x-datadog-trace-id": "797643193680388254",
+        },
+      };
+
+      const extractor = new HTTPEventTraceExtractor(tracerWrapper);
+
+      const traceContext = extractor.extract(payload);
+      expect(traceContext).not.toBeNull();
+
+      expect(spyTracerWrapper).toHaveBeenCalledWith({
+        "x-datadog-parent-id": "4726693487091824375",
+        "x-datadog-sampling-priority": "2",
+        "x-datadog-trace-id": "797643193680388254",
+      });
+
+      expect(traceContext?.toTraceId()).toBe("797643193680388254");
+      expect(traceContext?.toSpanId()).toBe("4726693487091824375");
+      expect(traceContext?.sampleMode()).toBe("2");
+      expect(traceContext?.source).toBe("event");
+    });
+
+    // The tracer is not handling mixed casing Datadog headers yet
+    it("extracts trace context from payload with mixed casing headers", () => {
+      mockSpanContext = {
+        toTraceId: () => "797643193680388254",
+        toSpanId: () => "4726693487091824375",
+        _sampling: {
+          priority: "2",
+        },
+      };
+      const tracerWrapper = new TracerWrapper();
+
+      const payload = {
+        headers: {
+          "X-Datadog-Parent-Id": "4726693487091824375",
+          "X-Datadog-Sampling-Priority": "2",
+          "X-Datadog-Trace-Id": "797643193680388254",
```

Comment:
Yes, it seems to me that the extractor can properly handle header with different cases.

Reply:
Yup!